test/Microsoft.Extensions.Caching.Memory.Tests/MemoryCacheSetAndRemoveTests.cs

```diff
@@ -121,13 +121,13 @@ public void SetOverwritesAndInvokesCallbacks()
             });
             result = cache.Set(key, value2, options2);
             Assert.Same(value2, result);
-            Assert.True(callback1Invoked.WaitOne(TimeSpan.FromSeconds(30)), "Callback1");
-            Assert.False(callback2Invoked.WaitOne(TimeSpan.FromSeconds(30)), "Callback2");
+            Assert.True(callback1Invoked.WaitOne(TimeSpan.FromSeconds(5)), "Callback1");
```

Comment:
The Assert.True timeouts need to be high so they don't timeout / become flaky on slow machines. Only lower the Assert.False timeouts to 1s.